pkg/rtc/room.go

```diff
@@ -322,16 +335,34 @@ func (r *Room) RemoveParticipant(identity string) {
 	}
 }
 
-func (r *Room) UpdateSubscriptions(participant types.Participant, trackIds []string, subscribe bool) error {
+func (r *Room) UpdateSubscriptions(
+	participant types.Participant,
+	trackIds []string,
+	participantTracks []*livekit.ParticipantTracks,
+	subscribe bool,
+) error {
 	// find all matching tracks
-	var tracks []types.PublishedTrack
+	tracks := make(map[string]types.PublishedTrack)
 	participants := r.GetParticipants()
-	for _, p := range participants {
-		for _, sid := range trackIds {
-			for _, track := range p.GetPublishedTracks() {
-				if sid == track.ID() {
-					tracks = append(tracks, track)
-				}
+	for _, trackSid := range trackIds {
+		for _, p := range participants {
+			track := p.GetPublishedTrack(trackSid)
+			if track != nil {
+				tracks[trackSid] = track
+				break
+			}
+		}
+	}
```

Comment:
optimising this a bit by asking participant for specific track sid and breaking as soon as one is found.